File: frontend/erpclink-web/src/app/features/clinics/models/clinic.models.ts
import { SimplePagedResult } from '../../../shared/models/api.models';

export interface CreateClinicRequest {
  code: string;
  name: string;
  description?: string | null;
  location?: string | null;
}

export interface UpdateClinicRequest {
  name: string;
  description?: string | null;
  location?: string | null;
}

export interface SearchClinicsParams {
  query?: string;
  code?: string;
  name?: string;
  branchId?: string;
  isActive?: boolean;
  sortBy?: string;
  sortDescending?: boolean;
  page?: number;
  pageSize?: number;
}

export interface ClinicDto {
  id: string;
  organizationId: string;
  branchId: string;
  code: string;
  name: string;
  description: string | null;
  location: string | null;
  isActive: boolean;
  createdAtUtc: string;
  createdBy: string | null;
  updatedAtUtc: string | null;
  updatedBy: string | null;
}

export interface ClinicListItemDto {
  id: string;
  code: string;
  name: string;
  location: string | null;
  isActive: boolean;
}

export type PagedClinicsResult = SimplePagedResult<ClinicListItemDto>;

File: src/Modules/Doctors/ErpClink.Modules.Doctors.Application/Clinics/Models/ClinicModels.cs
namespace ErpClink.Modules.Doctors.Application.Clinics.Models;

public sealed record CreateClinicRequest(
    string Code,
    string Name,
    string? Description,
    string? Location);

public sealed record UpdateClinicRequest(
    string Name,
    string? Description,
    string? Location);

public sealed record SearchClinicsRequest(
    string? Query,
    string? Code,
    string? Name,
    Guid? BranchId,
    bool? IsActive,
    string? SortBy,
    bool SortDescending = false,
    int Page = 1,
    int PageSize = 20);

public sealed record ClinicDto(
    Guid Id,
    Guid OrganizationId,
    Guid BranchId,
    string Code,
    string Name,
    string? Description,
    string? Location,
    bool IsActive,
    DateTime CreatedAtUtc,
    string? CreatedBy,
    DateTime? UpdatedAtUtc,
    string? UpdatedBy);

public sealed record ClinicListItemDto(
    Guid Id,
    string Code,
    string Name,
    string? Location,
    bool IsActive);

public sealed record PagedClinicsResult(
    IReadOnlyList<ClinicListItemDto> Items,
    int Page,
    int PageSize,
    int TotalCount);
